lib/query-emitter.js

```diff
@@ -310,17 +318,23 @@ QueryEmitter.prototype.onOp = function() {
   );
 };
 
-function getInserted(diff) {
-  var inserted = [];
+function getDiffItems(diff, currentValues) {
+  var removed = []
+  var inserted = []
   for (var i = 0; i < diff.length; i++) {
     var item = diff[i];
     if (item instanceof arraydiff.InsertDiff) {
       for (var j = 0; j < item.values.length; j++) {
         inserted.push(item.values[j]);
       }
+    } else if (item instanceof arraydiff.RemoveDiff) {
+      var values = currentValues.slice(item.index, item.index + item.howMany)
```

Comment:
Should I validate that `item.index` and `item.howMany` are populated with the correct types?

Reply:
If you mean, should we check `typeof foo === 'number'`, I don't think that's necessary, I don't recall needing to type-check arraydiff library results anywhere else.